core/src/main/java/jenkins/security/ApiTokenProperty.java

```diff
@@ -209,23 +209,27 @@ public boolean matchesPassword(String token) {
      */
     private boolean hasPermissionToSeeToken() {
         // Administrators can do whatever they want
-        if (SHOW_LEGACY_TOKEN_TO_ADMINS && Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
+        return checkPermission(SHOW_LEGACY_TOKEN_TO_ADMINS, user);
+    }
+
+    private static boolean checkPermission(boolean trustAdmins, User user) {
```

Comment:
:x: The name is too generic, so it has no meaning and brings confusion.

The two methods were in two different context with explicit name (at least from my PoV), with this generic method, we do not really know.

Proposals:
- checkSameUserOrAdmin
- isCurrentUserTheOwnerOrSuperUser
- any other proposition :)

Also changing `user` to `propertyOwner` could be good in terms of readability.

Reply:
The change from `user` to `propertyOwner` is easy for me to understand and do.

I'm going to have to chew on the method name.

Assuming I change that, the method suggestions so far don't fit.

Reply:
```suggestion
    private static boolean canCurrentUserControlObject(boolean trustAdmins, User propertyOwner) {
```